pkg/controller/aci_istio.go

```diff
@@ -145,6 +149,58 @@ func (cont *AciController) handleIstioUpdate(istiospec *istiov1.AciIstioOperator
 			return true
 		}
 	}
+
+	cfg, err := config.GetConfig()
+	iop := &iopv1alpha1.IstioOperator{}
+	scheme := runtime.NewScheme()
+	iopSchemeVersion := schema.GroupVersion{Group: "install.istio.io", Version: "v1alpha1"}
+	scheme.AddKnownTypes(iopSchemeVersion, &metav1.GetOptions{}, &metav1.UpdateOptions{}, iop)
+	istioclient, err := client.New(cfg, client.Options{Scheme: scheme})
+	if err != nil {
+		return true
+	}
+	err = istioclient.Get(context.TODO(), types.NamespacedName{Namespace: "istio-system", Name: "istiocontrolplane"}, iop)
+	if err != nil {
+		cont.log.Error("Error in Fetching istioctrlplaneCR")
+		log.Info(err)
+		return true
+	}
+	cont.log.Info("Setting Owner Reference for upstream istioCR with AciIstioOperator")
+	if !cont.isOwnerReferenceMarked(iop.ObjectMeta.OwnerReferences) {
+		iop.OwnerReferences = []metav1.OwnerReference{
+			*metav1.NewControllerRef(istiospec, istiov1.SchemeGroupVersion.WithKind("AciIstioOperator")),
+		}
+		err = istioclient.Update(context.TODO(), iop)
+		if err != nil {
+			cont.log.Error(err.Error())
+			return true
+		}
+	}
+
+	istioNamespaces := [...]string{"istio-system", "istio-operator"}
+	nsClient := kubeClient.CoreV1().Namespaces()
+	if nsClient == nil {
+		cont.log.Error("Error in Fetching nsClient")
+		return true // to requeue
+	}
+	for _, ns := range istioNamespaces {
+		istioNs, _ := nsClient.Get(context.TODO(), ns, metav1.GetOptions{})
```

Comment:
can you use the local cache to get the Namespace Object cont.namespaceIndexer.GetByKey(<nsname>) this way you can use local cache

Reply:
Key you can pass namespace name

Reply:
Good point; fixed.